vk/vk/ViewController/PushAnimator.swift

```diff
@@ -0,0 +1,62 @@
+// PushAnimator.swift
+// Copyright © Pavel V on RoadMap. All rights reserved.
+
+import UIKit
+
+/// PushAnimator is for custom screen transition animations
+final class PushAnimator: NSObject, UIViewControllerAnimatedTransitioning {
+    func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+        0.55
+    }
+
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+        guard
+            let source = transitionContext.viewController(forKey: .from),
+            let destination = transitionContext.viewController(forKey: .to)
+        else { return }
+
+        // Prepare initial position for both views
+        transitionContext.containerView.addSubview(destination.view)
+        destination.view.frame = source.view.frame
+        let xCoord = source.view.frame.width + source.view.frame.height / 2
+        let yCoord = source.view.frame.width / 2
+        // move destination view to the right and up
+        destination.view.center = CGPoint(x: xCoord, y: yCoord)
+        // rotate destination view counterclockwise
+        destination.view.transform = CGAffineTransform(rotationAngle: .pi / -2)
+
+        // Animations pool
+        UIView.animateKeyframes(
+            withDuration: transitionDuration(using: transitionContext),
+            delay: 0,
+            options: .calculationModePaced
+        ) {
+            // Move source view to the left and scale it down
+            UIView.addKeyframe(
+                withRelativeStartTime: 0,
+                relativeDuration: 0.6
+            ) {
+                let transform = CGAffineTransform(translationX: -200, y: 0)
+                let scale = CGAffineTransform(scaleX: 0.8, y: 0.8)
+                source.view.transform = transform.concatenating(scale)
+            }
+
+            // Move destination view to it's origin
```

Comment:
```suggestion
            // Move destination view to it's origin
```
вместо коментов лучше выносить в методы